src/debug/debugSupport.ts

```diff
@@ -45,3 +51,86 @@ export async function generateSourceMapUrl(functionText: string, generatedText:
   const sourceMapUrl = await sourceMap.generateSourceMapUrl(functionText, generatedText);
   return sourceMapUrl || generateSourceUrl();
 }
+
+export async function installConsoleHelpers(context: BrowserContextBase) {
+  if (!isDebugMode())
+    return;
+  const installInFrame = async (frame: Frame) => {
+    try {
+      const main = await frame._mainContext();
+      const injectedScript = await main.injectedScript();
+      await injectedScript.evaluate((injectedScript, parseSelectorString) => {
+        const parseSelector: (selector: string) => types.ParsedSelector = new Function('...args', 'return (' + parseSelectorString + ')(...args)') as any;
+        const checkSelector = (parsed: types.ParsedSelector) => {
+          for (const {name} of parsed.parts) {
+            if (!injectedScript.engines.has(name))
+              throw new Error(`Unknown engine "${name}"`);
```

Comment:
Can we do something about the nesting level here? I can't figure out where parseSelectorString is coming from.